phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -807,14 +810,39 @@ private void handleLocalIndexUpdates(TableName table,
             miniBatchOp.addOperationsFromCP(0, localUpdates.toArray(new Mutation[localUpdates.size()]));
         }
     }
+
+    private boolean isPartialUncoveredIndexUpdate(PhoenixIndexMetaData indexMetaData,
+            MiniBatchOperationInProgress<Mutation> miniBatchOp) throws IOException {
+        int indexedColumnCount = 0;
+        for (IndexMaintainer indexMaintainer : indexMetaData.getIndexMaintainers()) {
+            indexedColumnCount += indexMaintainer.getIndexedColumns().size();
+        }
+        Set<ColumnReference> indexedColumns = new HashSet<ColumnReference>(indexedColumnCount);
+        for (IndexMaintainer indexMaintainer : indexMetaData.getIndexMaintainers()) {
+            indexedColumns.addAll(indexMaintainer.getIndexedColumns());
+        }
+        for (int i = 0; i < miniBatchOp.size(); i++) {
+            if (miniBatchOp.getOperationStatus(i) == IGNORE) {
+                continue;
+            }
+            Mutation m = miniBatchOp.getOperation(i);
+            if (!this.builder.isEnabled(m)) {
+                continue;
+            }
+            for (ColumnReference indexedColumn : indexedColumns) {
+                if (m.get(indexedColumn.getFamily(), indexedColumn.getQualifier()).isEmpty()) {
+                    return true;
```

Comment:
nit: a comment here explaining how an empty column qualifier means the index update is partial